substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/config/ConfigurationParserUtils.java

```diff
@@ -124,12 +131,14 @@ public boolean tryAdvance(Consumer<? super URL> action) {
                 throw UserError.abort("Could not open %s: %s", url, e.getMessage());
             }
         }).sum();
-
         return parsedCount;
     }
 
     private static void doParseAndRegister(ConfigurationParser parser, Reader reader, String featureName, Object location, HostedOptionKey<LocatableMultiOptionValue.Strings> option) {
         try {
+            if (featureName.equals("dynamicClassLoading") && location instanceof URL) {
+                dynamicClassLoadingDirs.add((URL) location);
+            }
```

Comment:
Not clean and the static field will not work when there is more than one image build in the process (server mode). You might need to add a method to collect the locations and return them so you can use them in the feature's `afterRegistration` method.

Reply:
Removed the `dynamicClassLoadingDirs`. The original way is to fetch the dumped class file from all possible locations setting by `-H:DynamicClassesConfigurationFiles=` and `-H:DynamicClassesConfigurationResources=` , but the class file should be located in the `dynClasses` directory under the configuration file's directory. So we only need to follow the location of the currently parsed dynamicClass configuration file to look for the dumped class files.